### PR TITLE
Support for Discourse as comment system

### PR DESCRIPTION
**Pull request by Mte90**
### Pull Request Checklist

- [x] I’ve read the [guidelines for contributing](https://github.com/getnikola/nikola/blob/master/CONTRIBUTING.rst).
- [x] I updated AUTHORS.txt and CHANGES.txt (if the change is non-trivial) and documentation (if applicable).
- [ ] I tested my changes.

### Description

Add discourse as comment system, right now I didn't had time to test it
Ref: https://github.com/getnikola/nikola/issues/3689

### CODE REVIEW

File: nikola/data/themes/base/templates/comments_helper_discourse.tmpl [hunk 1]
```diff
@@ -0,0 +1,31 @@
+## -*- coding: utf-8 -*-
+
+<%! import json %>
+
+<%def name="comment_form(url, title, identifier)">
+    %if comment_system_id:
+        <div id="discourse-comments"></div>
+        <script type="text/javascript">
+            DiscourseEmbed = { discourseUrl: '${blog_url}',
```
Comment (Kwpolska) [inline nikola/data/themes/base/templates/comments_helper_discourse.tmpl:9]: This should probably be the Discourse forum’s URL, not the blog URL, please double-check. The Discourse URL can be placed in the comment_system_id setting (as done by e.g. isso), this should also be documented in manual.rst.

File: CHANGES.txt [hunk 2]
```diff
@@ -1,6 +1,11 @@
 New in master
 =============
 
+Features
+--------
+
+* Support for Discourse as comment system
```
Comment (Kwpolska) [inline CHANGES.txt:7]:
```suggestion
* Support for Discourse as comment system (Issue #3689)
```

File: nikola/data/themes/base/templates/comments_helper_discourse.tmpl (hunk 1 above)
Comment (Mte90) [reply nikola/data/themes/base/templates/comments_helper_discourse.tmpl]: you are right I have to test it asap

File: nikola/data/themes/base/templates/comments_helper_discourse.tmpl [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                d.src = DiscourseEmbed.discourseUrl + 'javascripts/embed.js';
+                (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(d);
+            })();
+        </script>
+    %endif
+</%def>
+
+<%def name="comment_link(link, identifier)">
+    %if comment_system_id:
+    <a href="${link}#discourse-comments" data-disqus-identifier="${identifier}">${messages("Comments")}</a>
```
Comment (Kwpolska) [inline nikola/data/themes/base/templates/comments_helper_discourse.tmpl:23]:
```suggestion
    <a href="${link}#discourse-comments">${messages("Comments")}</a>
```

File: nikola/data/themes/base/templates/comments_helper_discourse.tmpl [hunk 4]
```diff
@@ -0,0 +1,31 @@
+## -*- coding: utf-8 -*-
+
+<%! import json %>
```
Comment (Kwpolska) [inline nikola/data/themes/base/templates/comments_helper_discourse.tmpl:3]: Please remove this line.

File: nikola/data/themes/base/templates/comments_helper_discourse.tmpl [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+<%def name="comment_link(link, identifier)">
+    %if comment_system_id:
+    <a href="${link}#discourse-comments" data-disqus-identifier="${identifier}">${messages("Comments")}</a>
+    %endif
+</%def>
+
+
+<%def name="comment_link_script()">
+    %if comment_system_id:
```
Comment (Kwpolska) [inline nikola/data/themes/base/templates/comments_helper_discourse.tmpl:29]: This block should be empty, the `if` is unnecessary.

**Review by Kwpolska — APPROVED** [review]

**Comment by Kwpolska** [comment]
@Mte90 Please let us know when you’re done testing this so we can merge.

**Comment by Kwpolska** [comment]
(I changed the title, because we don’t really do Conventional Commits here.)

**Comment by Mte90** [comment]
Tested it and works :-D

You can test with https://gitlab.com/ItalianLinuxSociety/ils.org/-/tree/nikola?ref_type=heads
We will switch soon from jekyll to nikola :-)

**Comment by Kwpolska** [comment]
Thanks for contributing this!

**Comment by Mte90** [comment]
Thanks for your project :-)

**Comment by Mte90** [comment]
There are any plans for the new release?

**Comment by Kwpolska** [comment]
We don’t make releases that often, but you can always just copy the two template files into your site/venv.

**Comment by Mte90** [comment]
Cool I didn't know of that

**Pull request merged**